src/lib/core/components/visualizations/implementations/BarplotVisualization.tsx

```diff
@@ -371,12 +373,12 @@ function BarplotViz(props: VisualizationProps) {
       : [];
   }, [data]);
 
-  // use this to set all checked
-  useEffect(() => {
-    if (data != null) {
-      onCheckedLegendItemsChange(legendItems.map((item) => item.label));
-    }
-  }, [data, legendItems]);
+  // set checkedLegendItems
+  const checkedLegendItems = useMemo(() => {
```

Comment:
We could implement this as a custom hook, right?

Reply:
Yup, definitely.